src/vs/workbench/browser/actions/developerActions.ts

```diff
@@ -506,12 +508,104 @@ class RemoveLargeStorageEntriesAction extends Action2 {
 	}
 }
 
+let tracker: DisposableTracker | undefined = undefined;
+let trackedDisposables = new Set<IDisposable>();
+
+const DisposablesSnapshotStateContext = new RawContextKey<'started' | 'pending' | 'stopped'>('dirtyWorkingCopies', 'stopped');
+
+class StartTrackDisposables extends Action2 {
+
+	constructor() {
+		super({
+			id: 'workbench.action.startTrackDisposables',
+			title: { value: localize('startTrackDisposables', "Start Tracking Disposables"), original: 'Start Tracking Disposables' },
+			category: Categories.Developer,
+			f1: true,
+			precondition: ContextKeyExpr.and(DisposablesSnapshotStateContext.isEqualTo('pending').negate(), DisposablesSnapshotStateContext.isEqualTo('started').negate())
+		});
+	}
+
+	run(accessor: ServicesAccessor): void {
+		const disposablesSnapshotStateContext = DisposablesSnapshotStateContext.bindTo(accessor.get(IContextKeyService));
+		disposablesSnapshotStateContext.set('started');
+
+		trackedDisposables.clear();
+
+		tracker = new DisposableTracker();
+		setDisposableTracker(tracker);
+	}
+}
+
+class SnapshotTrackedDisposables extends Action2 {
+
+	constructor() {
+		super({
+			id: 'workbench.action.snapshotTrackedDisposables',
+			title: { value: localize('snapshotTrackedDisposables', "Snapshot Tracked Disposables"), original: 'Snapshot Tracked Disposables' },
+			category: Categories.Developer,
+			f1: true,
+			precondition: DisposablesSnapshotStateContext.isEqualTo('started')
+		});
+	}
+
+	run(accessor: ServicesAccessor): void {
+		const disposablesSnapshotStateContext = DisposablesSnapshotStateContext.bindTo(accessor.get(IContextKeyService));
+		disposablesSnapshotStateContext.set('pending');
+
+		trackedDisposables = new Set(tracker?.computeLeakingDisposables(1000)?.leaks.map(disposable => disposable.value));
+	}
+}
+
+class StopTrackDisposables extends Action2 {
```

Comment:
Since this is in OSS only, instead of an action you could insert an on top element with buttons into the `body`, that would also help avoid disposables in the command palette and be easier to use.

Reply:
For real tracking without quick pick I have started to use keybindings that ensure you really leak nothing.